qubes/tools/qubesd_query.py

```diff
@@ -75,20 +87,56 @@ async def qubesd_client(socket, payload, *args):
     finally:
         writer.close()
 
+# pylint: disable=too-many-return-statements
 def main(args=None):
     args = parser.parse_args(args)
     loop = asyncio.get_event_loop()
+    max_payload_size = 1024 if args.single_line else MAX_PAYLOAD_SIZE
+
+    if args.max_bytes is not None:
+        if args.max_bytes > max_payload_size:
+            parser.error('Maximum payload too large (max {})'
```

Comment:
64k is fine for the default limit, but may be a bit low for upper bound what is accepted at all. Anyway, preventing the user to raise the limit sounds pointless.

Reply:
Ah, right.